### PR TITLE
Close trio memory channel sender after the driver is quit

### LINKED ISSUE #13147: [🐛 Bug]: python program hangs forever when try to listen chrome bidi devtools in the background
### What happened?

I want to listen chrome bidi devtools to capture web page's http requests while performing certain click actions.

### How can we reproduce the issue?

``` python
import trio
import time
import threading
from selenium import webdriver

driver = webdriver.Chrome()


async def requests(connection):
    session, devtools = connection.session, connection.devtools
    pattern = map(devtools.fetch.RequestPattern.from_json, [{"urlPattern": "*"}])
    pattern = list(pattern)
    await session.execute(devtools.fetch.enable(patterns=pattern))

    return session.listen(devtools.fetch.RequestPaused)


async def at_request(event, connection):
    session, devtools = connection.session, connection.devtools
    print(event.request.url)
    return devtools.fetch.continue_request(request_id=event.request_id)


async def start_bidi_listener():
    async with driver.bidi_connection() as connection:
        session, devtools = connection.session, connection.devtools

        my_listener = await requests(connection)
        async with my_listener:
            async for event in my_listener:
                try:
                    await session.execute(await at_request(event=event, connection=connection))
                except Exception as e:
                    print(e)


def selenium_example():
    try:
        driver.get("https://www.google.com")
        threading.Thread(target=trio.run, args=(start_bidi_listener,)).start()
        time.sleep(6)
    finally:
        driver.quit()


if __name__ == "__main__":
    selenium_example()

```

### Relevant log output
python program hangs forever even when the driver is quit.

### Operating System

MacOS

### Selenium version

4.8.2

### What are the browser(s) and version(s) where you see this issue?

Chrome

### What are the browser driver(s) and version(s) where you see this issue?

Chrome

### Are you using Selenium Grid?

_No response_

### PR DESCRIPTION
**Pull request by jnhyperion**
fix: https://github.com/SeleniumHQ/selenium/issues/13147

the bug provide a thread version code snippet , this issue also exists in pure async io run.

``` python
import trio
from selenium import webdriver

driver = webdriver.Chrome()


async def requests(connection):
    session, devtools = connection.session, connection.devtools
    pattern = map(devtools.fetch.RequestPattern.from_json, [{"urlPattern": "*"}])
    pattern = list(pattern)
    await session.execute(devtools.fetch.enable(patterns=pattern))

    return session.listen(devtools.fetch.RequestPaused)


async def at_request(event, connection):
    session, devtools = connection.session, connection.devtools
    print(event.request.url)
    return devtools.fetch.continue_request(request_id=event.request_id)


async def start_bidi_listener():
    async with driver.bidi_connection() as connection:
        session, devtools = connection.session, connection.devtools

        my_listener = await requests(connection)
        async with my_listener:
            async for event in my_listener:
                try:
                    await session.execute(await at_request(event=event, connection=connection))
                except Exception as e:
                    print(e)


async def stop_driver():
    await trio.sleep(6)
    driver.quit()


async def selenium_example():
    driver.get("https://www.google.com")
    async with trio.open_nursery() as n:
        n.start_soon(start_bidi_listener)
        n.start_soon(stop_driver)


if __name__ == "__main__":
    trio.run(selenium_example)

```

before the fix, the code snippet will hang forever, after fix, it will exit properly.

### CODE REVIEW

File: py/selenium/webdriver/remote/webelement.py [hunk 1]
```diff
@@ -415,7 +416,7 @@ def find_element(self, by=By.ID, value=None) -> WebElement:
 
         return self._execute(Command.FIND_CHILD_ELEMENT, {"using": by, "value": value})["value"]
 
-    def find_elements(self, by=By.ID, value=None) -> list[WebElement]:
+    def find_elements(self, by=By.ID, value=None) -> List[WebElement]:
```
Comment (isaulv) [inline py/selenium/webdriver/remote/webelement.py:419]: What version of Python supports lower case `list`?

Comment (titusfortner) [reply py/selenium/webdriver/remote/webelement.py:419]: Looks like 3.9 and we still need to support 3.8

Comment (jnhyperion) [reply py/selenium/webdriver/remote/webelement.py:419]: this pr is created from my previous branch which also created another [pr](https://github.com/SeleniumHQ/selenium/pull/11735) long time ago, but looks like that pr is somehow reverted. That's why this change is here.

Comment (titusfortner) [reply py/selenium/webdriver/remote/webelement.py:419]: Ah, yeah, looks like it got removed when we updated syntax to 3.8

**Comment by titusfortner** [comment]
Thank you!

**Pull request merged**